src/core/Akka/Serialization/Serialization.cs

```diff
@@ -429,6 +429,11 @@ public object Deserialize(byte[] bytes, int serializerId, string manifest)
 
                 return serializer.FromBinary(bytes, type);
             }
+            catch (Exception e)
+            {
+                throw new SerializationException(
```

Comment:
This isn't really the right spot for this. Should be in the Akka.Remote serializer for wrapped payload support. We already handle this exception in the inner `try` block

Reply:
Actually, yes, it is the right spot for it, because the inner try..catch catches the exception thrown from failed manifest lookup, the outer exception catches the exception from the actual serializer deserialization. If it is not placed here then we have to actually put a try..catch on every single `FromBinary` function on every serializer.

Declaring it here actually makes debugging a very pleasant experience because all of the exceptions will be cascaded up the stack call from the top most call to the inner most calls when it is iterated.

Reply:
Basically, this fixes all exception rethrows for all serializers, not just Akka.Remote

Reply:
Ok

Reply:
Actually no - this is not a good fix.